### PR TITLE
Add docs for OPA access control

### PR DESCRIPTION
**Pull request by mosabua**
## Description

Converting readme from code PR into docs output and updating wording, grammar, content, and so on.

It would be great to have a few very simple rego scripts as example. Something like read only access for everyone, read only for specific catalogs only and users and groups, and maybe one or two more .. and then link to the relevant sections in the OPA docs for more. Any tips @vagaerg or @sbernauer ?

## Additional context and related issues

#19532

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

Release notes entry comes from code PR.

### CODE REVIEW

File: docs/src/main/sphinx/security/opa-access-control.md [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+All other fields in the request are identical to the non-batch endpoint.
+
+An OPA policy supporting batch operations should return a list containing the
+_indices_ of the items for which authorization is granted. Returning a `null`
+value or an empty list is equivalent and denies any access.
+
+You can batching support for policies that't originally do not support it:
+
+```text
+package foo
```
Comment (sbernauer) [inline docs/src/main/sphinx/security/opa-access-control.md:240]: As reported in Slack https://trinodb.slack.com/archives/C04HV35BJFR/p1704222429750449
```suggestion
package foo

import future.keywords.contains
```

**Review by sbernauer — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/security/opa-access-control.md (hunk 1 above)
Comment (mosabua) [reply docs/src/main/sphinx/security/opa-access-control.md:240]: Updated

File: docs/src/main/sphinx/security/opa-access-control.md [hunk 2]
```diff
@@ -0,0 +1,267 @@
+# Open Policy Agent access control
+
+Use [Open Policy Agent (OPA)](https://www.openpolicyagent.org/) as authorization
+engine for fine-grained access control to catalogs, schemas, tables, and more in
+Trino. Policies are defined in OPA, and Trino checks access control privileges
+in OPA.
+
+## Requirements
+
+* An running [OPA deployment](https://www.openpolicyagent.org/docs/latest/#running-opa)
```
Comment (vagaerg) [inline docs/src/main/sphinx/security/opa-access-control.md:10]:
```suggestion
* A running [OPA deployment](https://www.openpolicyagent.org/docs/latest/#running-opa)
```

nit :)

**Review by vagaerg — COMMENTED** [review]
LGTM other than the minor nitpick

Will work shortly on merging our work in https://github.com/bloomberg/trino/tree/add-open-policy-agent-with-docs/plugin/trino-opa/schema-docs here
That contains examples and full documentation for each operation

We've also built some JSON Schemas we could share, but that's potentially out of scope for this

**Comment by mosabua** [comment]
I fixed the nit and will leave PR as is for now. Lets see when we get the code PR merged. Then we can adjust the docs to the same level and add more later.

**Comment by dain** [comment]
This looks good.  I think we should remove duplicate stuff from the readme, so we don't have two places to maintain.

**Comment by mosabua** [comment]
@dain once the code PR is merged I will rebase and add the removal of the readme to this PR .. thats what we agreed with @vagaerg  .. I will do a last check to confirm I got it all..

**Review by dain — APPROVED** [review]
Minor comments, but otherwise looks good

File: docs/src/main/sphinx/security/opa-access-control.md [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+
+When request or response logging is enabled, details are logged at the `DEBUG`
+level under the `io.trino.plugin.opa.OpaHttpClient` logger. The Trino logging
+configuration must be updated to include this class, to ensure log entries are
+created.
+
+Note that enabling these options produces very large amounts of log data.
+
+(opa-permission-management)=
+### Permission management
```
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/security/opa-access-control.md:77]: I think I understand and will add that info.

[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/security/opa-access-control.md:77]: I am guessing the order would also matter then .. so the other system would have to be before.

File: docs/src/main/sphinx/security/opa-access-control.md [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+configure multiple access control configuration file paths in
+`etc/config.properties`:
+
+```properties
+access-control.config-files=etc/trino/file-based.properties,etc/trino/opa.properties
+```
+
+Configure each access-control system in the specified files.
+
+The following table lists all configuration properties for the OPA access control:
```
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/security/opa-access-control.md:36]: Done

File: docs/src/main/sphinx/security/opa-access-control.md [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+
+* - Name
+  - Description
+* - `opa.policy.uri`
+  - The **required** URI for the OPA endpoint, for example,
+    `https://opa.example.com/v1/data/allow`.
+* - `opa.policy.batched-uri`
+  - The **optional** URI for activating batch mode for certain authorization
+    queries where batching is applicable, for example
+    `https://opa.example.com/v1/data/batch`. Find more details in
```
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/security/opa-access-control.md:50]: Done

File: docs/src/main/sphinx/security/opa-access-control.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/security/opa-access-control.md:79]: I added another sentence .. hope that works.

**Pull request merged**